### ISSUE TITLE
Entity without relationships has unexpected behaviour

### ISSUE THREAD

**Issue by zeevkatz** [reyesoft/ngx-jsonapi#44]
One of my entities there's no relationships and it's seems like the api json service always looking for this property.
After fetching i'm get only the first entity and catch throw this error:
```
"__zone_symbol__currentTask": {
    "type": "microTask",
    "state": "notScheduled",
    "source": "Promise.then",
    "zone": "angular",
    "cancelFn": null,
    "runCount": 0
  }
```

**Comment by zeevkatz** [comment]
There something new about this issue?

**Comment by umbertix** [comment]
I've got my models without relationships declared atm. and I have no error whatsoever regarding this. Can you maybe provide some code example?

**Comment by zeevkatz** [comment]
This is my service:
```
@Injectable()
export class PermissionService extends Service<Permission> {
  public resource = Permission;
  public type = 'permissions';
  public schema: ISchema = {
  };
}
```

And this is my resource model:
```
export class Permission extends Resource {
  public attributes: {
    permissionType: string;
    deleted: boolean;
    created: string;
    groupId: number;
    updated: string;
  };
}
```

**Comment by umbertix** [comment]
I'm using the `.all()` but I convert it `.toPromise()` and I have no problems. Are you subscribing to it?

**Comment by umbertix** [comment]
I haven't been able to replicate your error, but to get another one. `can't convert undefined to object`. So yeah it seems like if the api doesn't return the relationships it does not like it too much 🤷‍♂️ 
Hopefully will be fix soon.

**Comment by pablorsk** [comment]
Hi @zeevkatz,

Your implementation looks fine, Can you provide code where is `all()` method? :eyes: 

@umbertix thanks for your help....

**Comment by zeevkatz** [comment]
Yeah, here is my .all() call below:
```
this.permissionService.all()
   .filter(data => !data.$is_loading)
   .map(data => new permissions.LoadSuccessAction(data.$toArray))
   .catch(error => Observable.of(new permissions.LoadFailedAction(error)))
```

**data** Includes only the first entity.
**error** return that object:
```
"__zone_symbol__currentTask": {
    "type": "microTask",
    "state": "notScheduled",
    "source": "Promise.then",
    "zone": "angular",
    "cancelFn": null,
    "runCount": 0
  }
```

**Comment by pablorsk** [comment]
`.all()` returns an observable, but without filter() function. This Observable is created vía `BehaviorSubject` from `rxjs/BehaviorSubject`. If you have a better idea, please post here recomendation. I see you have a better understanding about `observables`

![screenshot481](https://user-images.githubusercontent.com/938894/34717915-7e5ffb02-f513-11e7-8887-362c97ac8ea7.png)

**Comment by zeevkatz** [comment]
You can use filter pipe too, just import this operator from rxjs.
`import 'rxjs/add/operator/filter';`
But this is not the issue here, with or without the filter pipe.

**Comment by zeevkatz** [comment]
@pablorsk Somthing new about that issue?

**Comment by zeevkatz** [comment]
?

**Comment by maxi7587** [comment]
I have the same issue implementing Permissions. When I use ".all()", the API response is complete, but the library only gives me te first item... followed by this error in the console:

**Comment by pablorsk** [comment]
Great bug report @zeevkatz,

Its fixed on `1.0.0-rc.8`  :+1:
